.buildkite/pipeline.yml

```diff
@@ -0,0 +1,41 @@
+common-params:
+  &publish-android-artifacts-docker-container
+  docker#v3.8.0:
+    image: "public.ecr.aws/automattic/android-build-image:v1.1.0"
+    propagate-environment: true
+    environment:
+      # DO NOT MANUALLY SET THESE VALUES!
+      # They are passed from the Buildkite agent to the Docker container
+      - "AWS_ACCESS_KEY"
+      - "AWS_SECRET_KEY"
+
+steps:
+  - label: "Lint & Checkstyle"
+    key: "lint_and_checkstyle"
+    plugins:
+      - *publish-android-artifacts-docker-container
+    command: |
+      cp gradle.properties-example gradle.properties
+      ./gradlew lint checkstyle
+    artifact_paths:
+      - "**/build/reports/lint-results.*"
+      - "**/build/reports/checkstyle/checkstyle.*"
+  - label: "Test"
+    key: "test"
+    plugins:
+      - *publish-android-artifacts-docker-container
+    command: |
+      cp gradle.properties-example gradle.properties
+      ./gradlew test
+  # Publish tasks
+  - label: "Publish to s3"
+    depends_on:
+      - "lint_and_checkstyle"
+      - "test"
```

Comment:
👍  btw, I noticed that the [CircleCI config made the publication job only depend on test but _not_ on lint](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/74/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L53-L64), and I'm not sure why we allowed publication if the lint failed… I like this choice of depending on both better! 👌